cidr/cidr.go

```diff
@@ -110,3 +110,91 @@ func AddressCount(network *net.IPNet) uint64 {
 	prefixLen, bits := network.Mask.Size()
 	return 1 << (uint64(bits) - uint64(prefixLen))
 }
+
+//VerifyNetwork takes a list subnets and supernet (CIDRBlock) and verifies
+//none of the subnets overlap and all subnets are in the supernet
+//it returns an error if any of those conditions are not satisfied
+func VerifyNetwork(subnets []*net.IPNet, CIDRBlock *net.IPNet) error {
```

Comment:
I'll take the shorter one `VerifyNoOverlap` the only thing not clear in the name is hopefully obvious that it ensures the subnets are also included in the network (or supernet). I think that's obvious enough, unless you disagree.